index.js

```diff
@@ -121,7 +121,11 @@
 
   //       getTypeIdentifier :: TypeRepresentative -> String
   function getTypeIdentifier(Repr) {
-    return Repr[$$type] || Repr.name || 'Anonymous';
+    return (typeof Repr.prototype === 'object' ?
+            Repr.prototype[$$type] :
+            Repr[$$type]) ||
+           Repr.name ||
+           'Anonymous';
```

Comment:
@davidchambers 

This was a function that takes a Type Representative, and returns its type identifier. In the previous version of sanctuary-type-identifiers, that was as easy as returning its `@@type` property.

Now the best approach I could think of is to look at the prototype of the Repr. It won't always work, as many types are implemented as simple functions, rather than constructed classes.

My mistake was to rely on the `@@type` property as a public property. Instead I should have contributed this function to sanctuary-type-identifiers a long time ago, and made it your problem. :P

Nonetheless, what are your thoughts on this problem?

Reply:
Could you explain why it is necessary to get a type identifier from a type representative?

Reply:
Concurrify provides a function that, given a type representative and some functions, returns a new type representative. It automatically generates a type identifier under the same namespace that the given type representative was using for its type identity.